.github/workflows/apply-migrations.yaml

```diff
@@ -33,23 +33,32 @@ jobs:
       - name: Install dependencies
         run: npm install
         
-      - name: Generate migration preview
-        run: npm run db:generate
+      - name: Install Vercel CLI
+        run: npm install -g vercel
+        
+      - name: Link to Vercel project
+        run: vercel link --yes --token=${{ secrets.VERCEL_TOKEN }}
         env:
-          POSTGRES_URL: ${{ secrets.POSTGRES_URL }}
-          
-      - name: Run migration preview
+          VERCEL_TOKEN: ${{ secrets.VERCEL_TOKEN }}
+        
+      - name: Pull environment variables from Vercel
         run: |
-          npm run run-migrations -- --preview > migration-preview-full.txt 2>&1
-          sed -n '/=== Generated migration files ===/,$p' migration-preview-full.txt > migration-preview.txt
+          vercel env pull .env.vercel --environment=preview --token=${{ secrets.VERCEL_TOKEN }} --yes
+          POSTGRES_URL=$(grep NEON_URL .env.vercel | cut -d '=' -f2- | sed 's/^"//' | sed 's/"$//')
+          echo "POSTGRES_URL<<EOF" >> $GITHUB_ENV
+          echo "$POSTGRES_URL" >> $GITHUB_ENV
+          echo "EOF" >> $GITHUB_ENV
```

Comment:
This approach is not working, I have a ticket with Vercel investigating. In the meantime, use the Neon API to get the POSTGRES_URL for the branch DB dynamically